make/autoconf/libraries.m4

```diff
@@ -197,6 +199,30 @@ AC_DEFUN_ONCE([LIB_SETUP_MISC_LIBS],
   AC_SUBST(LIBDL)
   LIBS="$save_LIBS"
 
+  if test "x$OPENJDK_TARGET_OS" = "xbsd"; then
+    if test "x$OPENJDK_TARGET_OS_ENV" = "xbsd.openbsd"; then
+      ICONV_CFLAGS="-I/usr/local/include"
+      ICONV_LDFLAGS="-L/usr/local/lib"
+      ICONV_LIBS=-liconv
+    elif test "x$OPENJDK_TARGET_OS_ENV" = "xbsd.freebsd"; then
+      ICONV_CFLAGS=-DLIBICONV_PLUG
+      ICONV_LDFLAGS=
+      ICONV_LIBS=
+    else
+      ICONV_CFLAGS=
+      ICONV_LDFLAGS=
+      ICONV_LIBS=
+    fi
+  else
+    ICONV_CFLAGS=
+    ICONV_LDFLAGS=
+    ICONV_LIBS=-liconv
+  fi
+
+  AC_SUBST(ICONV_CFLAGS)
+  AC_SUBST(ICONV_LDFLAGS)
+  AC_SUBST(ICONV_LIBS)
```

Comment:
If you do this, I suggest you search the code base for existing mentions of `-liconv` and convert them to use ICONV_LIBS. As with pthreads, this can preferably be done separately ahead of the proper BSD port.